fix(backup): vault name may exceed 50 characters

jogold opened the pull request:
Remove `AWS::StackName` from the CDK generated vault name, it's already
included in the `uniqueId`.

BREAKING CHANGE: existing vaults that use a generated name will be replaced but
existing recovery points won't be lost. The default vault removal policy is
`RETAIN` and if it was set to `DESTROY` the deployment will fail because
vault with recovery points cannot be deleted.

Fixes #8627


----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject6AEA49D1-qxepHUsryhcu
* Commit ID: 7bef7d0e3decca0f0190d16555a5e1aaeaa0696b
* Result: SUCCEEDED
* [Build Logs](https://6sc75qyfl7.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7cf4503a-ddf4-44aa-a582-9769895ab94d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

MrArnoldPalmer commented:
@jogold I changed the "maturity" of the module to experimental which for more clear signal here. Also changed the note to be a "BREAKING CHANGE" annotation just to make sure users with existing generated name vaults see the message.

Let me know if you're okay with that.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject6AEA49D1-qxepHUsryhcu
* Commit ID: 3860f676355a25356e6cd11a41556564dd57416a
* Result: FAILED
* [Build Logs](https://6sc75qyfl7.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=01a3ef3d-73f1-4e3e-9e64-ca2f689d27f0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

jogold commented:
> @jogold I changed the "maturity" of the module to experimental which for more clear signal here. Also changed the note to be a "BREAKING CHANGE" annotation just to make sure users with existing generated name vaults see the message.
> 
> Let me know if you're okay with that.

👍 
(updated the stability banner in the README)

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject6AEA49D1-qxepHUsryhcu
* Commit ID: 613229262bad6c5385d5023d2e64bb9dc3acf707
* Result: SUCCEEDED
* [Build Logs](https://6sc75qyfl7.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2b2a1f3e-5c6c-4c61-8cca-074b3397751f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject6AEA49D1-qxepHUsryhcu
* Commit ID: 0b92bca92a0a7f19321a7306fb3cce3b83775714
* Result: SUCCEEDED
* [Build Logs](https://6sc75qyfl7.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=08fc4d48-d0c6-4df7-b97c-65b152e78788%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->
